pkg/common/cluster_state.go

```diff
@@ -349,11 +351,77 @@ func (i *ClusterState) readKeycloakIngressCurrentState(context context.Context,
 
 	err := controllerClient.Get(context, keycloakIngressSelector, keycloakIngress)
 	if err != nil {
-		if !errors.IsNotFound(err) {
+		if !apiErrors.IsNotFound(err) {
 			return err
 		}
 	} else {
 		i.KeycloakIngress = keycloakIngress.DeepCopy()
 	}
 	return nil
 }
+
+func (i *ClusterState) IsResourcesReady() (bool, error) {
```

Comment:
I think it's the right point to think about structuring our model. At the moment, every part of our model needs to following functionality:
- Return a new structure
- Return a reconciliated structure (for `UpdateAction`)
- Return a selector to it (so that we can check if it has been created or not)
- and a new functionality you just introduced - checking if its ready. 

Each piece of the functionality is different for each model type (whether it's a Service, a Route or anything). Perhaps we should create a JIRA to restructure it?